src/solve/slg/implementation/aggregate.rs

```diff
@@ -53,11 +51,12 @@ impl context::Aggregate<SlgContext> for SlgContext {
         // limited for that, but the on-demand SLG solver probably could
         // give us that information.
         let guidance = loop {
-            if subst.value.is_empty() || is_trivial(&subst) {
+            if subst.value.is_empty() || is_trivial(&subst)
+                || !simplified_answers.may_invalidate(&subst.value) {
```

Comment:
@nikomatsakis this seems to have been a bug! I believe we should return `Guidance::Definite(subst)` when `may_invalidate` is false.

(pinged on gitter but leaving it here too as documentation)